Src/LiquidProjections/ProjectionException.cs

```diff
@@ -0,0 +1,375 @@
+        /// and its inner exceptions with additional data.
+        /// The enriched exception should be rethrown via "throw" without arguments.
+        /// </summary>
+        /// <param name="innerException">An already thrown exception.</param>
+        /// <param name="setProperties">
+        /// The delegate that enriches the provided <see cref="ProjectionException"/> with additional data.
+        /// </param>
+        public static void ThrowNewOrEnrichExisting(Exception innerException, Action<ProjectionException> setProperties)
+        {
+            if (innerException == null)
+            {
+                throw new ArgumentNullException(nameof(innerException));
+            }
+
+            if (setProperties == null)
+            {
+                throw new ArgumentNullException(nameof(setProperties));
+            }
+
+            ProjectionException projectionException = FindProjectionException(innerException);
+
+            if (projectionException == null)
+            {
+                projectionException = new ProjectionException("Projector failed to project a transaction.", innerException);
+                setProperties(projectionException);
+                throw projectionException;
+            }
+
+            setProperties(projectionException);
+        }
+
+        private static ProjectionException FindProjectionException(Exception exception)
+        {
+            ProjectionException projectionException = null;
+            Exception currentException = exception;
+
+            while ((currentException != null) && (projectionException == null))
+            {
+                projectionException = currentException as ProjectionException;
+                currentException = currentException.InnerException;
+            }
+
+            return projectionException;
+        }
+
+        /// <summary>
+        /// Logs an exception that happened while projecting with additional data
+        /// about inner exceptions, call stacks, current projectors, transactions and events.
+        /// </summary>
+        /// <remarks>
+        /// Logging is handled by LibLog which may (or may not) need additional setup
+        /// depending on the logging library you use.
+        /// </remarks>
+        /// <param name="exception">The exception to log.</param>
+        /// <param name="logLevel">The logging level.</param>
+        public static void LogException(Exception exception, LogLevel logLevel)
+        {
+            if (exception == null)
+            {
+                throw new ArgumentNullException(nameof(exception));
+            }
+
+            LogProvider.For<ProjectionException>().Log(logLevel, () => BuildLogMessage(exception), exception);
+        }
+
+        private static string BuildLogMessage(Exception exception)
+        {
+            StringBuilder result = new StringBuilder();
+
+            Exception currentException = exception;
+
+            while (currentException != null)

```

Comment:
I don't think we need JSON conversion in LiquidProjections at all then. Exposing the objects themselves from the exception should be enough.